### PR TITLE
Accept colons instead of equals for named arguments

### PR DESCRIPTION
**Pull request by fabpot**
Refs #3475
Refs #3635

### CODE REVIEW

**Comment by smnandre** [comment]
(For information, i tested TwigComponent with this branch: all green)

File: doc/templates.rst [hunk 1]
```diff
@@ -210,11 +210,16 @@ built-in functions.
 Named Arguments
 ---------------
 
-Named arguments are supported in functions, filters and tests.
+Named arguments are supported in functions, filters, and tests.
+
+.. versionadded:: 3.12
+
+    Twig supports both ``=`` and ``:`` as separators between keys and values,
```
Comment (stof) [inline doc/templates.rst]:
```suggestion
    Twig supports both ``=`` and ``:`` as separators between argument names and values,
```

Comment (fabpot) [reply doc/templates.rst]: Changed

File: doc/templates.rst [hunk 2]
```diff
@@ -227,7 +232,7 @@ the values you pass as arguments:
 
     {# versus #}
 
-    {{ data|convert_encoding(from='iso-2022-jp', to='UTF-8') }}
+    {{ data|convert_encoding(from: 'iso-2022-jp', to: 'UTF-8') }}
```
Comment (smnandre) [inline doc/templates.rst:235]: As you replaced the `=` with `:` in this doc, does that mean using `:` will (sooner or later) be the "recommanded" syntax ?  

(something we wondered with @VincentLanglet to chose a potential "default choice" in Twig-CS-Fixer rule)

Comment (fabpot) [reply doc/templates.rst:235]: I think we will support both for the foreseeable future. I would recommend using `:` for a CS fixer though.

File: doc/coding_standards.rst [hunk 3]
```diff
@@ -105,3 +105,9 @@ standards:
              true
          {% endif %}
      {% endblock %}
+
+* Use ``:`` instead of ``=`` to separate argument names and values:
+
+  .. code-block:: twig
+
+     {{ data|convert_encoding(from: 'iso-2022-jp', to: 'UTF-8') }}
```
Comment (fabpot) [inline doc/coding_standards.rst:113]: cc @smnandre @VincentLanglet

**Pull request merged**